### PR TITLE
budget and forecasting

### PR DESCRIPTION
**Pull request by emsdelantar**

### CODE REVIEW

**Review by flowerinthenight — COMMENTED** [review]
@emsdelantar Looks like these RPCs are vendor neutral, so these formats should work for Azure and GCP. Since we don't really know that, it's probably better to create an AWS specific group results. Something like [this](https://github.com/alphauslabs/blueapi/blob/558e2d12b6baceccd00c5862d54297020bb38d4e/cost/v1/cost.proto#L301).

File: cost/v1/cost.proto [hunk 1]
```diff
@@ -172,6 +174,51 @@ service Cost {
...
+
+  // Fetches Budget Configuration for the specified account id
+  rpc GetBudgetConfigurationAccount(GetBudgetConfigRequest) returns (BudgetConfigResponse) {
+    option (google.api.http) = {
+      get: "/v1/{vendor}/budgets/configuration/{company_id}/{account_id}"
+    };
+  }
+
+  // Fetches Cost Forecasts for the specified Billing Group
+  rpc GetForecasts(GetForecastsRequest) returns (GetForecastsResponse) {
```
[the remark this answers is not in the document]
Comment (emsdelantar) [reply cost/v1/cost.proto:209]: GetForecasts() will not include current month's accumulated cost but will only show historical cost and up to three months forecast (3 months for now). MonthToDate(), however, will include current month actual cost vs forecasted cost vs budget. Maybe I need to change the function name but I don't think I can combine this with GetForecasts.

File: cost/v1/cost.proto [hunk 2]
```diff
@@ -172,6 +174,46 @@ service Cost {
       post: "/v1/{vendor}/accounts/{id}/tagcosts:read"
     };
   }
+
+  // Saves Budget Configuration to database
+  rpc CreateBudgetConfig(BudgetConfigRequest) returns (google.protobuf.Empty) {
```
Comment (flowerinthenight) [inline cost/v1/cost.proto:179]: @emsdelantar Please change to `CreateBudgetConfigRequest`

File: cost/v1/cost.proto [hunk 3]
```diff
@@ -172,6 +174,46 @@ service Cost {
...
+    option (google.api.http) = {
+      post: "/v1/{vendor}/budgets/configuration"
+      body: "budgetConfig"
+    };
+  }
+
+  // Updates Budget Configuration in database
+  rpc UpdateBudgetConfig(BudgetConfigRequest) returns (google.protobuf.Empty) {
```
Comment (flowerinthenight) [inline cost/v1/cost.proto:187]: @emsdelantar Please change to `UpdateBudgetConfigRequest`

**Review by flowerinthenight — APPROVED** [review]

**Pull request merged**